### ISSUE TITLE
Wrap nullable properties with Nullable wrapper

### ISSUE THREAD

**Issue by sionzeecz** [ikkentim/SampSharp#293]
When we work with API we do not except to be null occurred from the API.
But for an example: "src/SampSharp.GameMode/Events/ClickPlayerEventArgs.cs"

Property:

	/// <summary>
	///     Gets the clicked player.
	/// </summary>
	public BasePlayer ClickedPlayer { get; private set; }

In "src/SampSharp.GameMode/BaseMode.callbacks.cs"

	[Callback]
	internal bool OnPlayerClickPlayer(int playerid, int clickedplayerid, int source)
	{
		OnPlayerClickPlayer(BasePlayer.FindOrCreate(playerid),
			new ClickPlayerEventArgs(
				clickedplayerid == BasePlayer.InvalidId ? null : BasePlayer.FindOrCreate(clickedplayerid),
				(PlayerClickSource) source));

		return true;
	}

Line:

	clickedplayerid == BasePlayer.InvalidId ? null : BasePlayer.FindOrCreate(clickedplayerid),

Means in listeners will be ClickedPlayer null.

To me, these nullable variables should be wrapped with Nullable (aka Optional) wrapper.

Then it should be:

	if(args.ClickedPlayer.Has) {
		var clickedPlayer = args.ClickedPlayer.Get;
	}

If there will be wrapper for these nullable types then developers knows there can null occur and protect it.

**Comment by ikkentim** [comment]
No. C# simply allows null values and I'll use that feature here the  same way -1/InvalidId indicates no player in the SAMP API. When the next C# settles in I'll gladly add nullable indicators but I won't create some weird nullable wrapper myself.

**Comment by sionzeecz** [comment]
Well but there are methods where you cannot guarantee the player will exist.
If I write a listener for an event PlayerDeath. I don't expect the killer to be null.

I just mean the four event above. If you are not interested in wrapper solution then you should add a comment to event doc. Exactly to the EventArgs something like: "Killer can be null" **because no one expect a null from the API**. 

This is like having an empty array but not returning the empty array but null because the array is empty.

Otherwise you can add check methods to these events. For an example: IsKillerAvailable?

**Comment by ikkentim** [comment]
I could add a note in the docs of the property.
